apps/web/src/components/NavContext/NavContextContainer.tsx

```diff
@@ -21,6 +21,19 @@ const NAV_ROLES_BY_PRIVILEGE = [
 
 export type NavRole = (typeof NAV_ROLES_BY_PRIVILEGE)[number];
 
+export const isNavRole = (x: unknown): x is NavRole => {
+  switch (x) {
+    case "applicant":
+    case "manager":
+    case "community":
+    case "admin":
+      return true;
+    default:
+      return false;
+  }
+};
+// x && typeof x === 'string' ? NAV_ROLES_BY_PRIVILEGE.includes(x) :
```

Comment:
Why didn't you go with this commented out version?  It's not very dry to have to remember to add new roles in two different places.

Reply:
Oh I forgot to remove this :sweat_smile:  It wasn't playing nice with typescript. I was trying to do something like [this](https://stackoverflow.com/a/51529486), but that was also causing errors with our [no-explicit-any rule](https://typescript-eslint.io/rules/no-explicit-any/). So, I on what's above for now.

Reply:
I tried to rewrite it a bit and this is what I got:
```ts
export const isNavRole = (x: unknown): x is NavRole => {
  if (typeof x === "string") {
    return NAV_ROLES_BY_PRIVILEGE.includes(x as NavRole);
  }
  return false;
};
```

It seems to work well, except for the null. Do you really want null in the navrole list?  That seems very weird to me.  You can just make a variable nullable if you want to include that.

![image](https://github.com/user-attachments/assets/e10451f0-d322-4b2f-aa0e-68df37164e78)

Reply:
Maybe a more sane cast:

Reply:
Ugh, this would be a bigger refactor than I want to deal with right now.